addons/microsoft_account/controllers/main.py

```diff
@@ -12,10 +13,12 @@ class MicrosoftAuth(http.Controller):
     @http.route('/microsoft_account/authentication', type='http', auth="public")
     def oauth2callback(self, **kw):
         """ This route/function is called by Microsoft when user Accept/Refuse the consent of Microsoft """
-        state = json.loads(kw['state'])
+        state = json.loads(kw.get('state', '{}'))
         dbname = state.get('d')
         service = state.get('s')
         url_return = state.get('f')
+        if (not dbname or not service or (kw.get('code') and not url_return)):
```

Comment:
actually dbname seems not used ?

Reply:
Done. I have removed `dbname` from here.
Thanks!